### DISCUSSION TITLE
Most performant way of displaying 2d array of integer values

### DISCUSSION THREAD

**Discussion by ramiromagno** [raysan5/raylib#2635] [Q&A]
Say I have a 2d array of integer values, and that I'd like to show it as an "image"/"texture" using OpenGL. For each integer value should correspond a color.

What is the most performant way of dispatching that data into the GPU and show it on the screen?

**Comment by raysan5** [comment]
@ramiromagno what do you mean? draw a rectangle? draw a text value?

**Reply by ramiromagno** [reply]
Hi @raysan5,

I want to draw a pixel for each integer value. Not sure if that pixel should actual be a quad... I am just starting with OpenGL, so excuse my ignorance. But I guess that I want to push the data stored in a 2d array `int **`, and, according to a color palette, i.e., a mapping from integer values to colors, draw a texture corresponding to that array.

My use case would be the following: running a spatial simulation, where space is 2D and is encoded by the array. Example: perform cellular automata simulations, https://en.wikipedia.org/wiki/Cellular_automaton#/media/File:Gas_velocity.gif.

My focus is on performance, i.e., to have as little as possible overhead on the communication between the array and the texture (?).

**Reply by SleepingNerd** [reply]
Hey Ramiro lmk if you found something interesting about this, dealing with the same problem atm.

**Reply by ramiromagno** [reply]
Hey @SleepingNerd: I have not found a good solution, no.

**Reply by SleepingNerd** [reply]
I currently use UpdateTexture for this, I store an extra color array because of this though. I'd rather store indexes that correspond to colors, but this is fine for now.
